BUG: Flaky test `test_truncation_specialized_op`

### Describe the issue:

It failed in #6304 and on main: https://github.com/pymc-devs/pymc/actions/runs/3665014447/jobs/6195812078

### Reproduceable code example:

```python
pytest -v pymc/tests/distributions/test_truncated.py::test_truncation_specialized_op
```


### Error message:

```shell
FAILED pymc/tests/distributions/test_truncated.py::test_truncation_specialized_op[observed] - assert False
 +  where False = isinstance(<pytensor.tensor.elemwise.Elemwise object at 0x7fdf5924e1a0>, TruncatedNormalRV)
 +    where <pytensor.tensor.elemwise.Elemwise object at 0x7fdf5924e1a0> = Elemwise{identity}(AdvancedIncSubtensor{inplace=False,  set_instead_of_inc=True}.0).op
 +      where Elemwise{identity}(AdvancedIncSubtensor{inplace=False,  set_instead_of_inc=True}.0) = xt.owne
```


### PyMC version information:

main

### Context for the issue:

_No response_

--- a/pymc/tests/distributions/test_truncated.py
+++ b/pymc/tests/distributions/test_truncated.py
@@ -84,9 +84,7 @@ def test_truncation_specialized_op(shape_info):
                 dist=x,
                 lower=5,
                 upper=15,
-                observed=np.empty(
-                    100,
-                ),
+                observed=np.zeros(100),
             )
         else:
             raise ValueError(f"Not a valid shape_info parametrization: {shape_info}")

